# HG changeset patch
# User Ted Campbell <tcampbell@mozilla.com>
# Date 1664986722 0
# Node ID de384dc334c4b86f9735c35e745230c8fe9ee785
# Parent  2a3ee45a29093ce07286fe054fd14a22f8ddf94c
Bug 1792159 - Add missing includes to AtomicOperationsGenerated.h r=sfink

On certain build configurations involving GCC, we use additional annotations
that are defined in Attributes.h so we must include the header first. For
consistency, include on all platforms since we use this header safely in many
places already.

Differential Revision: https://phabricator.services.mozilla.com/D158595

diff --git a/js/src/jit/GenerateAtomicOperations.py b/js/src/jit/GenerateAtomicOperations.py
--- a/js/src/jit/GenerateAtomicOperations.py
+++ b/js/src/jit/GenerateAtomicOperations.py
@@ -702,16 +702,18 @@ HEADER_TEMPLATE = """\
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #ifndef jit_AtomicOperationsGenerated_h
 #define jit_AtomicOperationsGenerated_h
 
 /* This file is generated by jit/GenerateAtomicOperations.py. Do not edit! */
 
+#include "mozilla/Attributes.h"
+
 namespace js {
 namespace jit {
 
 %(contents)s
 
 } // namespace jit
 } // namespace js
 

